fiona/ogrext.pyx

```diff
@@ -665,11 +665,9 @@ cdef class Session:
                 raise ValueError("Null field definition")
 
             key_c = OGR_Fld_GetNameRef(cogr_fielddefn)
+            if key_c == NULL:
+                raise ValueError("Empty field name at index: %s" % i)
```

Comment:
"Empty" isn't correct because you intend to allow empty strings, yes? How about "Field name is NULL at index"?

I think it's highly unlikely we'll see a NULL field name, but we do need to avoid passing a NULL on in our code.

Reply:
Agreed :+1: